main/webservices/api/test_api.php

```diff
@@ -2,116 +2,550 @@
+        $modelCourseList = array_keys(SessionManager::get_course_list_by_session_id($modelSessionId));
+        $newCourseList = array_keys(SessionManager::get_course_list_by_session_id($newSessionId));
+        $this->assertSame($modelCourseList, $newCourseList);
+
+        // assert the current url was set on the new session
+        if (api_is_multiple_url_enabled()) {
+            $urls = UrlManager::get_access_url_from_session($newSessionId);
+            $this->assertSame(1, count($urls));
+            $this->assertSame(api_get_current_access_url_id(), intval($urls[0]['access_url_id']));
+        }
+
+        // clean up
+        foreach($COURSE_CODES as $code) {
+            CourseManager::delete_course($code);
+        }
+        SessionManager::delete($modelSessionId);
+        SessionManager::delete($newSessionId);
+        $promotion->delete($promotionId);
+        $career->delete($careerId);
+        $extraFieldModel->delete($extraFieldId);
+        $extraFieldModel->delete($secondExtraFieldId);
+        if (api_is_multiple_url_enabled()) {
+            UrlManager::delete($urlId);
+        }
     }
 
-    $jsonResponse = json_decode($response->getBody()->getContents());
+    public function testSubscribeUserToSessionFromUsername()
+    {
+        $ACTION = 'subscribe_user_to_session_from_username';
+
+        // create a session
+        $sessionId = SessionManager::create_session(
+            'Session to subscribe'.time(),
+            '2019-01-01 00:00', '2019-08-31 00:00',
+            '2019-01-01 00:00', '2019-08-31 00:00',
+            '2019-01-01 00:00', '2019-08-31 00:00',
+            null, null
+        );
+
+        // create a user
+        $loginName = 'tester'.time();
+        $userId = UserManager::create_user('Tester', 'Tester', 5, 'tester@local', $loginName, 'xXxxXxxXX');
 
-    if ($jsonResponse->error) {
-        throw new Exception('User not created because : ' . $jsonResponse->message);
+        // create another user and subscribe it to the session
+        $anotherUserId = UserManager::create_user('Tester 2', 'Tester 2', 5, 'tester2@local', $loginName.'t2', 'xXxxX');
+        SessionManager::subscribeUsersToSession($sessionId, [$anotherUserId]);
+
+        // call the webservice to subscribe the first user to the session
+        $subscribed = $this->boolean($ACTION, [ 'sessionId' => $sessionId, 'loginName' => $loginName ] );
+        $this->assertTrue($subscribed);
+
+        // assert we now have two users subscribed to the session
+        $sessionRelUsers = Database::getManager()
+            ->getRepository('ChamiloCoreBundle:SessionRelUser')
+            ->findBy([ 'session' => $sessionId ]);
+        $this->assertSame(2, count($sessionRelUsers));
+
+        // clean up
+        UserManager::delete_users([ $userId, $anotherUserId ]);
+        SessionManager::delete($sessionId);
     }
-    return $jsonResponse->data[0];
-}
 
-$apiKey = authenticate();
+    public function testGetSessionFromExtraField()
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The variable $SECOND_EXTRA_FIELD_NAME is not named in camelCase.](https://app.codacy.com/app/chamilo/chamilo-lms/pullRequest?prid=4554450)